expo-boomboomapp/src/api/SpotifyApiService/SpotifyApiServiceI.ts

```diff
@@ -1,14 +1,17 @@
 export type Track = {
-  popularity?: string;
+  popularity?: number;
   name: string; // song name
   trackId: string;
   album?: string; // album name
   image?: string; // song image
-  artistName: string;
+  artistName?: string;
+  uri: string;
 };
 
 export abstract class SpotifyApiServiceI {
   abstract fetchTop5Tracks(): Promise<Track[]>;
 
-  abstract fetchTracksNyName(name?: string): Promise<Track[]>;
+  abstract fetchTracksNyName(
+    name?: string
+  ): Promise<{ data: Track[]; status: boolean }>;
```

Comment:
I think it's bad to adapt the frontend according to the backend has it is now, here by exemple the `status` field is not needed because the info is already in the http response. 

If the response is 200 we know already if the backend is good or not. We need to play with response http status and not with respondy body status. 

I will rework the backend. In the meantime with can develop frontend separately from the backend with mocks class. And at this end we will merge the two part. 
What do you think ?

Reply:
I think it's good to have a body with data and other things, it's more scalable. But yes the status can be manage differently